vault/ui_custom_messages/message.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package uicustommessages
+
+import "time"
+
+var allowedMessageTypes = [2]string{
```

Comment:
This is a nit and feel free to ignore, but I think it makes sense to add the type names as separate constants, e.g.

```
var bannerMessageType = "banner"
var modalMessageType = "modal"

var allowedMessageTypes = [2]string{
  bannerMessageType, modalMessageType,
}